### DISCUSSION TITLE
Troubles with creating a config template for HP Aruba switch

### DISCUSSION THREAD

**Discussion by Wildeagle23** [netbox-community/netbox#16242] [Help Wanted!]
Hello,

For the last couple days i have tried to create a config template for a aruba switch, but get stuck on something.

If you have a Cisco switch it wil define vlans under the interface.
But on a Aruba switch it defines the tagged and untagged ports under the vlan interface.

Now have I created a couple normal interfaces and a vlan interface under a test device in Netbox.
But I can't seen to get all the ports used as tagged and untagged under a vlan.

I have used this configuration till now but I am stuck at this point:
{# Creating variables for (un-)tagged vlans #}
{% set tagged_vlans = {} %}
{%- for interface in device.interfaces.all() %}
    {%- if "vlan" in interface.name %}
        {% set vlan_name = "tagged_" ~ interface.name %}
        {% set _ = tagged_vlans.update({vlan_name: interface}) %}
        {% set untaggedvlan_name = "untagged_" ~ interface.name %}
        {% set _ = untagged_vlans.update({untaggedvlan_name: interface}) %}
    {%- endif %}
{%- endfor %}

{%- for vlan_name, interface in tagged_vlans.items() %}
    {# Now I can use variables as tagged_vlan1, tagged_vlan2 etc. #}
    {{ vlan_name }}: {{ interface }}
{%- endfor %}

{%- for untaggedvlan_name, interface in untagged_vlans.items() %}
    {# Now I can use variables as untagged_vlan1, untagged_vlan2 etc. #}
    {{ untaggedvlan_name }}: {{ interface }}
{%- endfor %}

{# For test use only!!! #}
{% set untagged_vlan10 = test %}

{# From here configuration #}

{% for interface in device.interfaces.all() %}
{%- if "vlan" in interface.name | lower %}
{%- else %}
	name "{{ interface.label }}"
{%- endif %}
{% endfor %}

{% for interface in device.interfaces.all() %}
{%- if "vlan" in interface.name | lower %}
interface {{ interface.name }}
name "{{ interface.label }}"

{% set tagged = '{{ "tagged_" + interface.name }}' %}
{{ tagged }}
untagged {{ "untagged_" + interface.name }}
{%- endif %}
{% endfor %}

What I am trying to do is create a variable based on the name of a vlan interface which I have accomplished.
But now I need to recreate that variable at the vlan interface, is there a way to do that based on the name if the interface?

I hope somebody can help me.

**Comment by candlerb** [comment]
> But on a Aruba switch it defines the tagged and untagged ports under the vlan interface.

Here are two versions of a prototype template I wrote for a Mikrotik device, which may be similar to what you're trying to do.  Try them and see if either is helpful.

```
{%- set dif = device.interfaces.exclude(type="virtual") %}
{%- for v in device.site.vlans.all() %}
{%- set ift = ((dif.filter(tagged_vlans=v)|list) + (dif.filter(mode="tagged-all")|list)) | unique | sort(attribute="name") %}
{%- set ifu = dif.filter(untagged_vlan=v) | sort(attribute="name") -%}
{%- if ift or ifu %}
/interface bridge vlan add bridge=bridge vlan-ids={{v.vid}}
{%- if ift %} tagged={{ ift|join(",") }}{% endif %}
{%- if ifu %} untagged={{ ifu|join(",") }}{% endif %}
{%- endif %}
{%- endfor %}
```

```
{%- for v in device.site.vlans.all() %}
{%- set ift = ((v.interfaces_as_tagged.exclude(type="virtual").filter(device=device)|list)
             + (device.interfaces.filter(mode="tagged-all")|list))|sort(attribute="name") %}
{%- set ifu = v.interfaces_as_untagged.exclude(type="virtual").filter(device=device)|sort(attribute="name") %}
{%- if ift or ifu %}
/interface bridge vlan add bridge=bridge vlan-ids={{v.vid}}
{%- if ift %} tagged={{ ift|join(",") }}{% endif %}
{%- if ifu %} untagged={{ ifu|join(",") }}{% endif %}
{%- endif %}
{%- endfor %}
```

There is a variation of this for Zyxel switches in #12573
